instrumentation/opentelemetry-instrumentation-logging/src/opentelemetry/instrumentation/logging/__init__.py

```diff
@@ -0,0 +1,125 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=empty-docstring,no-value-for-parameter,no-member,no-name-in-module
+
+import logging  # pylint: disable=import-self
+from os import environ
+from typing import Callable
+
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.logging.constants import (
+    _MODULE_DOC,
+    DEFAULT_LOGGING_FORMAT,
+)
+from opentelemetry.instrumentation.logging.environment_variables import (
+    OTEL_PYTHON_LOG_CORRELATION,
+    OTEL_PYTHON_LOG_FORMAT,
+    OTEL_PYTHON_LOG_LEVEL,
+)
+from opentelemetry.trace import (
+    INVALID_SPAN,
+    INVALID_SPAN_CONTEXT,
+    get_current_span,
+    get_tracer_provider,
+)
+
+__doc__ = _MODULE_DOC
+
+LEVELS = {
+    "debug": logging.DEBUG,
+    "info": logging.INFO,
+    "warning": logging.WARNING,
+    "error": logging.ERROR,
+}
+
+
+class LoggingInstrumentor(BaseInstrumentor):  # pylint: disable=empty-docstring
+    __doc__ = """An instrumentor for stdlib logging module.
+
+    This instrumentor injects tracing context into logging records and optionally sets the global logging format to the following:
+
+    .. code-block::
+
+        {default_logging_format}
+
+    Args:
+        tracer_provider: Tracer provider instance that can be used to fetch a tracer.
+        set_logging_format: When set to True, it calls logging.basicConfig() and sets a logging format.
+        logging_format: Accepts a string and sets it as the logging format when set_logging_format
+            is set to True.
+        log_level: Accepts one of the following values and sets the logging level to it.
+            logging.INFO
+            logging.DEBUG
+            logging.WARN
+            logging.ERROR
+            logging.FATAL
+
+    See `BaseInstrumentor`
+    """.format(
+        default_logging_format=DEFAULT_LOGGING_FORMAT
+    )
+
+    _old_factory = None
+
+    def _instrument(self, **kwargs):
+        service_name = ""
+        provider = kwargs.get("tracer_provider", None) or get_tracer_provider()
+        resource = provider.resource if provider else None
+        if resource:
+            service_name = resource.attributes.get("service.name")
+
```

Comment:
@owais I have been going through the codebase to see places we are relying on the current global tracer provider and in particularly where it doesn't fit with the multiple tracer providers usecase. This part of code is getting service name from the global tracer provider. Is this still something we want to do here?

Side question: Does this log-trace correlation also apply to logs emitted by opentemetry python packages?

Reply:
> Is this still something we want to do here?

No, I think we should update this to extract it from span (resource) just like the span and trace ID.

> Side question: Does this log-trace correlation also apply to logs emitted by opentemetry python packages?
Yes. As it stands today, once enabled, this feature will apply to all logs produced by the python process.